### ISSUE TITLE
Static proptypes aren't found when using typescript-eslint/parser

### ISSUE THREAD

**Issue by remleduff** [yannickcr/eslint-plugin-react#3183]
I have a mixed js/ts codebase that I'm slowly adding more linting to.

For the js code, estlint-plugin-react correctly doesn't warn about proptypes when using babel/eslint-parser for the js code, but I'd like to switch all my code to use typescript/eslint-parser so that my type information can be used more broadly.

However, when I switch parsers, every proptype of the following form seems to not be understood:

```
class Foo extends React.Component {
  static propTypes = {
    values: PropTypes.object.isRequired,
    updateValue: PropTypes.func.isRequired,
  };
```

I'd really hate to lose the proptypes warning, but I also can't have 500+ warnings, so I'm a little stuck for the moment.

Alternatively, is there a way to configure eslintrc so that one parser is used for almost everything, but then another parser is used with just one rule? I haven't managed so far because the 'overrides' don't seem to like having the same glob in different configurations.

**Comment by ljharb** [comment]
Can you share the exact package name and version of the parser you're using, as well as your eslint config?

Yes, you can indeed use "overrides" to use the TS parser for `.ts` files, and the default parser for `.js` files, for example.

**Comment by remleduff** [comment]
Yes, as I said, "overrides" works fine if I'd like to completely use one parser for each type of file. 

What I want though is to use typescript-eslint/parser for all files (with react/prop-types turned off since it doesn't seem to work). And then use babel/eslint-parser for only js files (with only react/prop-types turn on).

I'm using:
```
typescript": "^4.5.4"
@typescript-eslint/eslint-plugin": "^5.10.0",
@typescript-eslint/parser": "^5.10.0",
```

.esllintrc
```
const lint_strict_any = "off";

// Unfortunately, running into an issue here where react/proptypes doesn't
// seem to work for static proptypes when using typescript-eslint/parser
// rather than babel/eslint-parser
const lint_proptypes = "error"

module.exports = {
  root: true,
  extends: [
    "eslint:recommended",
    "plugin:jest/recommended",
    "plugin:react/recommended",
    "plugin:react-hooks/recommended",
    "plugin:@typescript-eslint/recommended",
    "plugin:@typescript-eslint/recommended-requiring-type-checking",
  ],
  env: {
    "browser": true,
    "es6": true,
    "jest": true,
  },
  globals: {
    "process": true,
    "devToolsExtension": true,
    "module": true,
  },
  parser: "@typescript-eslint/parser",
  parserOptions: {
    ecmaVersion: 14,
    tsconfigRootDir: __dirname,
    project: ['./tsconfig.json'],
    babelOptions: {
            configFile: "./.babelrc"
    },
  },
  settings: {
    react: {
      version: "detect"
    }
  },
  rules: {
    "no-prototype-builtins": "warn",
    "quotes": ["error", "single", { allowTemplateLiterals: true }],
    "prefer-const": "error",
    "no-var": "error",
    "comma-dangle": ["error", "only-multiline"],
    "jsx-quotes": ["error", "prefer-double"],
    "semi": ["error", "always"],

    "react/prop-types": lint_proptypes,
    //"react/jsx-curly-brace-presence": "error",

    // Not supported on our version of node
    "@typescript-eslint/no-loss-of-precision": "off",

    "@typescript-eslint/no-use-before-define": ["error", { "functions": false }],
    "@typescript-eslint/no-unused-vars": ["error", { "varsIgnorePattern": "[iI]gnored" }],

    // We would like to use all of these eventually, but for now it's too much
    "@typescript-eslint/no-unsafe-argument": lint_strict_any,
    "@typescript-eslint/no-unsafe-member-access": lint_strict_any,
    "@typescript-eslint/no-unsafe-call": lint_strict_any,
    "@typescript-eslint/no-unsafe-assignment": lint_strict_any,
    "@typescript-eslint/restrict-template-expressions": lint_strict_any,
    "@typescript-eslint/no-unsafe-return": lint_strict_any,
    "@typescript-eslint/restrict-plus-operands": lint_strict_any,
    "@typescript-eslint/restrict-template-expressions": lint_strict_any,
    "@typescript-eslint/no-floating-promises": lint_strict_any,
  }
};
```

**Comment by ljharb** [comment]
You can certainly do that, but the prop-types rule works fine with the TS parser - although we're not testing on v5, i believe. v4 should certainly work. Can you try that?

**Comment by remleduff** [comment]
Actually, I think I spoke too soon. Originally my @typescript-eslint/parser was older, and I upgraded it when you asked me to paste the version.

I initially thought it was still broken after upgrading, but it seems to actually be working now. Thanks!
